py/torch_tensorrt/_Input.py

```diff
@@ -364,9 +364,25 @@ def example_tensor(
                     )
 
                 if isinstance(self.shape, dict):
-                    return torch.rand(self.shape[optimization_profile_field]).to(
-                        dtype=self.dtype.to(torch.dtype, use_default=True)
-                    )
+                    if (
+                        self.dtype == dtype.u8
+                        or self.dtype == dtype.i8
+                        or self.dtype == dtype.i32
+                        or self.dtype == dtype.i64
+                    ):
+                        type = self.dtype.to(torch.dtype, use_default=True)
+                        min_value = torch.iinfo(type).min
+                        max_value = torch.iinfo(type).max
+                        return torch.randint(
+                            min_value,
+                            max_value,
+                            self.shape[optimization_profile_field],
+                            dtype=type,
+                        )
```

Comment:
casted int random generated all zero. Change to torch.randint() for integer random values
e.g
>>> torch.rand((2,3)).to(dtype=dtype.int32.to(torch.dtype, use_default=True))
tensor([[0, 0, 0],
        [0, 0, 0]], dtype=torch.int32)

Reply:
Do we need to do the same changes for static as well? https://github.com/pytorch/TensorRT/pull/2878/files#diff-12d8a77266209f04943c413a5e36d6cd060036c57857782955c48ea9ea39acc9R346-R349

Reply:
Yes I think we need it. Updated in new commit